tgui/packages/tgui/interfaces/NtosMessenger/index.tsx

```diff
@@ -0,0 +1,342 @@
+import { Box, Button, Icon, Section, Stack, Input, TextArea, Dimmer, Divider } from '../../components';
+import { useBackend, useLocalState } from '../../backend';
+import { createSearch } from 'common/string';
+import { BooleanLike } from 'common/react';
+import { NtosWindow } from '../../layouts';
+import { SFC } from 'inferno';
+
+import { NtChat, NtMessenger, NtPicture } from './types';
+import { ChatScreen } from './ChatScreen';
+
+type NtosMessengerData = {
+  can_spam: BooleanLike;
+  is_silicon: BooleanLike;
+  owner: NtMessenger | null;
+  saved_chats: Record<string, NtChat>;
+  messengers: Record<string, NtMessenger>;
+  sort_by_job: BooleanLike;
+  alert_silenced: BooleanLike;
+  alert_able: BooleanLike;
+  sending_and_receiving: BooleanLike;
+  open_chat: string;
+  stored_photos?: NtPicture[];
+  selected_photo_path: string | null;
+  on_spam_cooldown: BooleanLike;
+  virus_attach: BooleanLike;
+  sending_virus: BooleanLike;
+};
+
+export const NtosMessenger = (_props: any, context: any) => {
+  const { act, data } = useBackend<NtosMessengerData>(context);
+  const {
+    is_silicon,
+    saved_chats,
+    stored_photos,
+    selected_photo_path,
+    open_chat,
+    messengers,
+    sending_virus,
+  } = data;
+
+  let content: JSX.Element;
+  if (open_chat !== null) {
+    content = (
+      <ChatScreen
+        chat={saved_chats[open_chat]}
+        temp_msgr={messengers[open_chat]}
+        storedPhotos={stored_photos}
+        selectedPhoto={selected_photo_path}
+        onReturn={() => act('PDA_viewMessages', { ref: null })}
+        isSilicon={is_silicon}
+        sendingVirus={sending_virus}
+      />
+    );
+  } else {
+    content = <ContactsScreen />;
+  }
+
+  return (
+    <NtosWindow width={600} height={850}>
+      <NtosWindow.Content>{content}</NtosWindow.Content>
+    </NtosWindow>
+  );
+};
+
+const ContactsScreen = (_props: any, context: any) => {
+  const { act, data } = useBackend<NtosMessengerData>(context);
+  const {
+    owner,
+    alert_silenced,
+    alert_able,
+    sending_and_receiving,
+    saved_chats,
+    messengers,
+    sort_by_job,
+    can_spam,
+    is_silicon,
+    virus_attach,
+    sending_virus,
+  } = data;
+
+  const [searchUser, setSearchUser] = useLocalState<string>(
+    context,
+    'searchUser',
+    ''
+  );
+
+  // you are not ready for this garbage o7
+
+  let savedChatsArray = Object.values(saved_chats);
+  savedChatsArray.sort((a, b) => b.unread_messages - a.unread_messages);
+
+  if (searchUser.length > 0) {
+    const chatSearch = createSearch(
+      searchUser,
+      (chat: NtChat) => chat.recp.name + chat.recp.job
+    );
+
+    savedChatsArray = savedChatsArray.filter(chatSearch);
```

Comment:
You could then pipe this onto the `sorted` call by appending `.filter(createSearch(etc))`. Don't worry about the searchUser length, createSearch should handle that quickly.